firmware-binaries/sim-tests/c_registerwb_test/build.rs

```diff
@@ -0,0 +1,41 @@
+// SPDX-FileCopyrightText: 2025 Google LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use memmap_generate::build_utils::standard_memmap_build;
+
+fn main() {
+    // Generate memory.x linker script
+    standard_memmap_build("RegisterWb.json", "DataMemory", "InstructionMemory");
+
+    // Use C headers from auto-generated bittide-hal-c
+    let hal_c_headers = bittide_hal_c::generated_dir();
+
+    // Include directory for the Bittide HAL
+    let hal_c_include = bittide_hal_c::manual_include_dir();
+
+    // Source directory for the Bittide HAL
+    let hal_c_src = bittide_hal_c::manual_source_dir();
+
+    // Compile C code with clang (has built-in RISC-V support)
+    cc::Build::new()
+        .compiler("clang")
+        .target("riscv32imc-unknown-none-elf")
+        .flag("-march=rv32imc")
+        .flag("-mabi=ilp32")
+        .flag("-nostdlib")
+        .opt_level(3)
+        .include(&hal_c_headers)
+        .include(&hal_c_include)
+        .file("src/main.c")
+        .files(
+            std::fs::read_dir(&hal_c_src)
+                .expect("Failed to read HAL C source directory")
+                .filter_map(Result::ok)
+                .map(|entry| entry.path())
+                .filter(|path| path.extension().is_some_and(|ext| ext == "c")),
+        )
```

Comment:
there's `bittide_hal_c::manual_source_files()` which does that same thing and is already used in other build scripts for C crates.